### ISSUE TITLE
Make all output paths absolute

### ISSUE THREAD

**Issue by Southclaws** [Southclaws/pawn#247]
http://forum.sa-mp.com/showpost.php?p=3976383&postcount=17

Errors and warnings have different path styles depending on whether they are in the primary compilation unit (not sure what else to call this, the file you "compile from", the root of your include tree?) This affects editor integration for "jump-to-error" tools.

Errors in the main script look like this:

```
gamemode(1) : fatal error 100: cannot read from file: "inc"
```

Whereas errors in included files look like this:

```
/home/southclaws/pawn/test/inc(1) : fatal error 100: cannot read from file: "inc2"
```

This is something I fixed in sampctl by processing the output but regular pawncc users can't really do anything about it.

**Comment by Zeex** [comment]
AFAIK it's relative to the current working directory by default *or* to the directory passed in the `-D` flag, if any. Can you access cwd and compile flags from VS Code?

**Comment by Southclaws** [comment]
Yes you can access all of that info in vscode and most other editors afaik. The issue is the fact that there's two output modes, it would be better if it was either *all* absolute or *all* relative, not a mix of both. There's already an open issue in the vscode repo to support both but it doesn't seem to be getting pushed up their priority list unfortunately.

**Comment by Zeex** [comment]
Changing the compiler because of a bug in one editor doesn't seem right. It should be the other way around.

Whether pawnc outputs relative or absolute paths depends in what paths you use on the command line. If you pass relative paths, it will use relative paths in errors and warnings. If you pass absolute paths, it will use absolute paths.

```
PS E:\> pawncc -i"..\include" test\1.pwn
Pawn compiler 3.10.6                    Copyright (c) 1997-2006, ITB CompuPhase

..\include\2.inc(1) : warning 235: public function lacks forward declaration (symbol "other_func")
..\include\2.inc(2) : error 017: undefined symbol "a"
test\1.pwn(4) : error 017: undefined symbol "func"
```

```
PS E:\> pawncc -i"E:\include" E:\test\1.pwn
Pawn compiler 3.10.6                    Copyright (c) 1997-2006, ITB CompuPhase

E:\include\2.inc(1) : warning 235: public function lacks forward declaration (symbol "other_func")
E:\include\2.inc(2) : error 017: undefined symbol "a"
E:\test\1.pwn(4) : error 017: undefined symbol "func"
```

So in other to have always absolute or always relative paths you have to use absolute or relative paths in include directories supplied via `-i` and input file names.

**Comment by Southclaws** [comment]
To be clear, it's not a bug in the editors at all, they just do what they are documented to do.

The issue is just the inconsistency. I'll relay this info back to the forum though and some people should get some use out of it. It's not always the best course of action to supply full paths to `-i` as you might want to keep build settings portable either between your own machines, developers in your team's machines and CI services. I still think it would be useful to tell the compiler to *always* output full paths via a simple flag.
